src/main/scala/edu/colorado/plv/cuanto/scoot/ir/package.scala

```diff
@@ -0,0 +1,46 @@
+package edu.colorado.plv.cuanto.scoot
+
+/**
+  * Created by Jared on 6/13/2017.
+  */
+package object ir {
+  implicit def convertAddExpr(dt: soot.jimple.AddExpr) = new AddExpr(dt)
+
+  implicit def convertBinopExpr(dt: soot.jimple.BinopExpr) : BinopExpr = {
+    if(dt.isInstanceOf[soot.jimple.AddExpr]) convertAddExpr(dt.asInstanceOf[soot.jimple.AddExpr])
+    else if(dt.isInstanceOf[soot.jimple.DivExpr]) convertDivExpr(dt.asInstanceOf[soot.jimple.DivExpr])
+    else if(dt.isInstanceOf[soot.jimple.MulExpr]) convertMulExpr(dt.asInstanceOf[soot.jimple.MulExpr])
+    else if(dt.isInstanceOf[soot.jimple.SubExpr]) convertSubExpr(dt.asInstanceOf[soot.jimple.SubExpr])
+    else throw new Exception("unsupported BinopExpr")
+  }
```

Comment:
@jpw234 Note on the visitor implementation in Soot

- The Jimple nodes implement the Switchable interface: https://soot-build.cs.uni-paderborn.de/doc/soot/soot/jimple/parser/node/Switchable.html

The interface defines the apply method: by calling the apply method, you start to recursively visit the Jimple AST.
The apply method takes as input an object of Switch type and the visit calls the methods implemented in the switch object to handle different type of AST nodes.

The idea is that you override the methods in the Switch object and they will be automatically called "at the right time" during the visit of the AST. 

- Switch is an interface: https://soot-build.cs.uni-paderborn.de/doc/soot/soot/util/Switch.html
There are different sub interfaces that defines the method that will be called when visiting different kinds of nodes.

For example, the StmtSwitch interface (https://soot-build.cs.uni-paderborn.de/doc/soot/soot/jimple/StmtSwitch.html) defines the methods that are called when the visit reach a statement, and you have methods that you have to override like 	caseAssignStmt(AssignStmt stmt) that will be called when the visit reach an AssignStmt node.

There are examples of the visitor implemented in the wild, e.g. JayHorn
https://github.com/jayhorn/jayhorn/blob/44a946341e3ac6d62ee982228d40aeb56516a304/soottocfg/src/main/java/soottocfg/soot/visitors/SootStmtSwitch.java